### PR TITLE
Split consensus parameters into their own Config struct

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

**Comment by kchulj** [comment]
The changes are not reflected in some unittests. (tests failing)

**Comment by linked0** [comment]
The variables like `quorum_shuffle_interval`, `tx_payload_max_size`, `tx_payload_fee_factor` exists in the test code for NodeConfig as follows.
https://github.com/bpfkorea/agora/blob/ed5bc28c8a421ebabd0b71b3b796e19bed12fe0b/source/agora/common/Config.d#L464-L471

For validator_cycle, a new `ConsensusConfig` variable is needed in files for the networking test like Base.d.

**Comment by kchulj** [comment]
I think there should be a clean-up of `agora.consensus.data.Params` as the next step. Thoughts?

File: source/agora/common/Config.d [hunk 1]
```diff
@@ -148,6 +156,21 @@ public struct NodeConfig
...
+    /// The duration between requests for retrieving the latest blocks
+    /// from all other nodes
+    public Duration block_catchup_interval = 20.seconds;
+
+    /// The new block timestamp has to be greater than the previous block timestamp,
+    /// but less than current time + block_timestamp_tolerance
+    public Duration block_timestamp_tolerance = 60.seconds;
+}
+
+/// Loosely matches `agora.consensus.data.Params : ConsensusParans`
```
Comment (kchulj) [inline source/agora/common/Config.d]: Typo :P
```suggestion
/// Loosely matches `agora.consensus.data.Params : ConsensusParams`
```

**Comment by Geod24** [comment]
> I think there should be a clean-up of `agora.consensus.data.Params` as the next step. Thoughts?

Yep, although it's low-prio. I just wanted to avoid `NodeConfig` getting out of hand (also I have to add some fields to it for another PR).

File: source/agora/common/Config.d (hunk 1 above)
Comment (Geod24) [reply source/agora/common/Config.d]: Good eye

File: source/agora/common/Config.d [hunk 2]
```diff
@@ -455,23 +452,77 @@ node:
...
+private ConsensusConfig parseConsensusConfig (Node* node, in CommandLine cmdln)
+{
+    const validator_cycle = get!(uint, "validator", "validator_cycle")(cmdln, node);
+    const max_quorum_nodes = get!(uint, "validator", "max_quorum_nodes")(cmdln, node);
+    const quorum_threshold = get!(uint, "validator", "quorum_threshold")(cmdln, node);
+    const quorum_shuffle_interval = get!(uint, "validator", "quorum_shuffle_interval")(cmdln, node);
+    const tx_payload_max_size = get!(uint, "validator", "tx_payload_max_size")(cmdln, node);
+    const tx_payload_fee_factor = get!(uint, "validator", "tx_payload_fee_factor")(cmdln, node);
+    const validator_tx_fee_cut = get!(ubyte, "validator", "validator_tx_fee_cut")(cmdln, node);
+    const payout_period = get!(uint, "validator", "payout_period")(cmdln, node);
```
Comment (Geod24) [inline source/agora/common/Config.d]: 🤦

**Comment by Geod24** [comment]
> std.conv.ConvOverflowException@/opt/hostedtoolcache/dc/ldc2-65b0fb88/x64/ldc2-65b0fb88-linux-x86_64/bin/../import/std/conv.d(2337): Overflow in integral conversion

~Yeah maybe `validator_tx_fee_cut` shouldn't be a `ubyte`.~

Oh it's a percentage, oops

**Comment by Geod24** [comment]
Green

**Review by AndrejMitrovic — APPROVED** [review]

**Review by kchulj — APPROVED** [review]

File: tests/system/node/5/config.yaml [hunk 3]
```diff
@@ -14,14 +14,12 @@ node:
   timeout: 100
   # Path to the data directory (if the path doesn't exist it will be created)
   data_dir: .cache
-
-  tx_payload_max_size: 1024
```
Comment (linked0) [inline tests/system/node/5/config.yaml:18]: I think the `tx_payload_max_size` is missing. Is it intended?

Comment (Geod24) [reply tests/system/node/5/config.yaml:18]: Yes. It was set to its default value.

**Pull request merged**